### ISSUE TITLE
Separate resource for gateway_rest_api resource policy

### ISSUE THREAD

**Issue by zoltan-toth-mw** [hashicorp/terraform-provider-aws#5834]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Create an _aws_api_gateway_rest_api_resource_policy_ resource so in the resource policy I can use the api id of _the aws_api_gateway_rest_api_ resource, unlike now with the inlined policy option.

### New or Affected Resource(s)

* aws_api_gateway_rest_api_resource_policy

### Potential Terraform Configuration

```hcl
resource "aws_api_gateway_rest_api_resource_policy" "example" {
  api_gateway_rest_api_id = "${api_gateway_rest_api.example.id}"
 
  policy = <<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": "*",
            "Action": "execute-api:Invoke",
            "Resource": "arn:aws:execute-api:region:accound-id:api-id/${api_gateway_rest_api.example.id}/*",
            "Condition" : {
                "StringNotEquals": {
                    "aws:SourceVpce": "${data.terraform_remote_state.example.api_gateway_id}"
                }
            }
        }
    ]
}
EOF

}
```

**Comment by bflad** [comment]
Hi @zoltan-toth-mw 👋 

Just to understand the use case, is this just to workaround manually filling in the ARN pieces for the `Resource` portion of the policy?

**Comment by zoltan-toth-mw** [comment]
@bflad basically yes, as you can't do something like ${self.id}.

**Comment by itunesnosignup** [comment]
@bflad The following issue is the same - and explains it a bit better: https://github.com/terraform-providers/terraform-provider-aws/issues/5549
This would be one of the many possible solutions along with:
- Allow for the use of self.arn within this resource
- an auto-expansion of "Action": "execute-api:Invoke" - this is done via AWS automatically so when it's saved it's saved as the arn (therefore causing a side by side diff of the two policies)

**Comment by itunesnosignup** [comment]
As a follow up - if you change the policy to be the full expanded value e.g.
"Resource": "arn:aws:execute-api:<region>:<account_id>:<api_id>/*/POST/*", 
Then it no longer detects it as a change. The problem is api_id isn't a usable or known value at creation.

**Comment by bostrowski13** [comment]
so is there anything being done about this or #5549 ?

**Comment by flungo** [comment]
This change would also allow for the `aws_api_gateway_deployment` to trigger a redeployment when the policy changes (using the hash of the the `aws_api_gateway_rest_api` or even just it's `policy` attribute causes an inconsistent final plan as the value gets updated immediately after applying with the expanded value for `execute-api:*/*/*`).

This seems like the best solution for #5549 and is similar to how policies are managed on other AWS resources.

**Comment by bflad** [comment]
Support for this new resource has been merged and will release with version 3.16.0 of the Terraform AWS Provider, later this week. Thank you to @DrFaust92 for the implementation. 👍
